include/vigra/linear_solve.hxx

```diff
@@ -93,6 +95,41 @@ T determinantByLUDecomposition(MultiArrayView<2, T, C1> const & a)
     return det;
 }
 
+template <class T, class C1>
+typename NumericTraits<T>::Promote
+determinantByMinors(MultiArrayView<2, T, C1> const & mat)
+{
+    typedef typename NumericTraits<T>::Promote PromoteType;
+    MultiArrayIndex m = rowCount(mat);
+    MultiArrayIndex n = columnCount(mat);
+    vigra_precondition(
```

Comment:
What happened to the precondition on the promote type's signedness? I think it is still needed, preferably by means of std::static_assert(), because it can be checked at compile time.